### PR TITLE
adds absolute urls for serializers with image/file fields

### PR DESCRIPTION
**Pull request by yadavankit**
Passing the `request` object in `context` to `ChallengeSerializer` and `SubmissionSerializer` was enough for building absolute urls for media files. There was no need to define `SerializerMethodField` and call `build_absolute_uri` method on `request`.
@deshraj please review

### CODE REVIEW

**Comment by deshraj** [comment]
Good catch @yadavankit. Can you please fix the build errors?

**Comment by yadavankit** [comment]
@deshraj TravisCI Build runs into an `AssertionError` because of -
`AssertionError: [OrderedDict([('participant_team', 75), ('participant_team_name', u'Participant Team for Challenge'), ('execution_time', u'None'), ('challenge_phase', 39), ('created_by', 178), ('status', u'submitted'), ('input_file', 'http://testserver/media/test_annotations/test_sample_file.txt'), ('stdout_file', None), ('stderr_file', None), ('submitted_at', u'2017-02-09T23:06:19.623460Z')])] != [{'status': u'submitted', 'stderr_file': None, 'participant_team_name': 'Participant Team for Challenge', 'participant_team': 75, 'input_file': '/media/test_annotations/test_sample_file.txt', 'execution_time': u'None', 'challenge_phase': 39, 'stdout_file': None, 'created_by': 178, 'submitted_at': '2017-02-09T23:06:19.623460Z'}]`
Test case written should be altered I guess

**Comment by deshraj** [comment]
Yes, the test needs to be changed.

**Comment by yadavankit** [comment]
@deshraj since the unit-tests run on the `testserver`, I have appended `http://testserver` to input file for passing the build.

File: tests/unit/jobs/test_views.py [hunk 1]
```diff
@@ -348,7 +348,7 @@ def test_get_challenge_submissions(self):
                 'challenge_phase': self.submission.challenge_phase.pk,
                 'created_by': self.submission.created_by.pk,
                 'status': self.submission.status,
-                'input_file': self.submission.input_file.url,
+                'input_file': 'http://testserver' + self.submission.input_file.url,
```
Comment (deshraj) [inline tests/unit/jobs/test_views.py]: @yadavankit can we use string formatting instead of using `+` operator?

Comment (yadavankit) [inline tests/unit/jobs/test_views.py]: Yeah sure 👍

**Comment by deshraj** [comment]
Great Job @yadavankit. :-)

**Pull request merged**